### PR TITLE
Improve interactive docs

### PR DESCRIPTION
**Pull request by eshaben**
- [x] Fill out descriptions (source, arguments, returns) for each RPC method
- [x] Add example for each RPC method
- [x] Create mobile first layout with collapsible side nav

This PR addresses a small part of #680

### CODE REVIEW

File: src/chains/ethereum/ethereum/src/api.ts [hunk 1]
```diff
@@ -1787,31 +2633,47 @@ export default class EthereumApi implements types.Api {
...
    * should remain unlocked for. Set to 0 to disable automatic locking.
-   * @returns true if it worked. Throws an error if it did not.
+   * @returns `true` if it worked. Throws an error or returns `false` if it did not.
+   * @example
+   * ```javascript
+   * // generate an account
+   * const passphrase = "passphrase";
+   * const newAccount = await provider.send("personal_newAccount", [passphrase] );
+   * const isLocked = await provider.send("personal_unlockAccount", [newAccount, passphrase] );
+   * console.log(isLocked);
```
Comment (davidmurdoch) [inline src/chains/ethereum/ethereum/src/api.ts]:
```suggestion
   * const isUnlocked = await provider.send("personal_unlockAccount", [newAccount, passphrase] );
   * console.log(isUnlocked);
```

File: src/chains/ethereum/ethereum/src/api.ts [hunk 2]
```diff
@@ -1822,17 +2684,37 @@ export default class EthereumApi implements types.Api {
...
 
+  // TODO: example doesn't return correct value
   /**
    * Validate the given passphrase and submit transaction.
    *
-   * The transaction is the same argument as for eth_sendTransaction and
+   * The transaction is the same argument as for `eth_sendTransaction` and
    * contains the from address. If the passphrase can be used to decrypt the
-   * private key belonging to tx.from the transaction is verified, signed and
+   * private key belogging to `tx.from` the transaction is verified, signed and
```
Comment (davidmurdoch) [inline src/chains/ethereum/ethereum/src/api.ts]:
```suggestion
   * private key belonging to `tx.from` the transaction is verified, signed and
```

File: src/chains/ethereum/ethereum/src/api.ts [hunk 3]
```diff
@@ -1822,17 +2684,37 @@ export default class EthereumApi implements types.Api {
     );
   }
 
+  // TODO: example doesn't return correct value
```
Comment (davidmurdoch) [inline src/chains/ethereum/ethereum/src/api.ts]: Do we know what is wrong here? I think everything is working and this line can be removed.

File: src/chains/ethereum/ethereum/src/api.ts [hunk 4]
```diff
@@ -1858,9 +2740,69 @@ export default class EthereumApi implements types.Api {
...
+   *
+   * @param transaction - The transaction call object as seen in source.
+   * @returns The raw, signed transaction.
+   * @example
+   * ```javascript
+   * const [to] = await provider.request({ method: "eth_accounts", params: [] });
+   * const passphrase = "passphrase";
+   * const from = await provider.send("personal_newAccount", [passphrase] );
+   * await provider.request({ method: "eth_subscribe", params: ["newHeads"] });
+   * const signedTx = await provider.request({ method: "personal_signTransaction", params: [{ from, to}, passphrase] });
```
Comment (davidmurdoch) [inline src/chains/ethereum/ethereum/src/api.ts]:
```suggestion
   * const signedTx = await provider.request({ method: "personal_signTransaction", params: [{ from, to }, passphrase] });
```

**Review by davidmurdoch — CHANGES_REQUESTED** [review]

File: src/chains/ethereum/ethereum/src/api.ts (hunk 3 above)
Comment (MicaiahReid) [reply src/chains/ethereum/ethereum/src/api.ts]: I was able to take the returned txHash to retrieve a receipt, so I think this is working as expected. Removing line.

**Review by davidmurdoch — APPROVED** [review]

**Pull request merged**